src/models/information/_filters.js

```diff
@@ -75,6 +78,24 @@ const URBAN_ZONES = [
   "10-17",
 ]
 
+const provinceFilter = function() {
+  return provinces.reduce(
```

Comment:
You can use `_.mapValues` because reducing this way is O(n<sup>2</sup>) while `_.mapValues` is O(n)

Reply:
How is reduce O(n^2)?

Reply:
@dtinth I tried to use `_.mapValues` but it returns an object that has an index as a key
<img width="485" alt="Screen Shot 2562-03-24 at 15 23 57" src="https://user-images.githubusercontent.com/26592279/54876830-13022e80-4e49-11e9-8a2e-9002b3f151eb.png">
Do you how `_.mapValues` can return an object that has a province name as a key?

Reply:
```js
import { flow, keyBy, mapValues } from 'lodash/fp'

flow(
  keyBy(p => p.name),
  mapValues(province => ...)
)(provinces)
```

Reply:
better import individual function instead of importing all functions from lodash `_`.
also this use the functional flavor of it which is more modular, and use `flow` to chain them together.

Reply:
@kristw Thank you so much krubb, It works, Is it better than Array.reduce?

Reply:
> How is reduce O(n^2)?

`reduce` by itself is O(n), but `{ ...acc }` is also O(n), so they combine to become O(n^2) krub.

Reply:
Thank you for your description, It's my new knowledge.